### ISSUE TITLE
CS1712: false diagnostic for inheritdoc

### ISSUE THREAD

**Issue by anreton** [dotnet/roslyn#55799]
**Brief description:**

Describe your suggestion here.

**Languages applicable:**

Is your analyzer C# only? VB Only? Or Both?

**Code example that the analyzer should report:**

```csharp
	/// <summary>
	/// Represents a bar abstraction.
	/// </summary>
	/// <typeparam name="T">
	/// The type of the T.
	/// </typeparam>
	public interface IBar<T>
		where T : notnull
	{
		/// <summary>
		/// Gets the T value.
		/// </summary>
		/// <value>
		/// A value.
		/// </value>
		public T ValueOfT{ get; }
	}

	/// <summary>
	/// Represents a foo abstraction.
	/// </summary>
	/// <typeparam name="V">
	/// The type of the V.
	/// </typeparam>
	/// <inheritdoc cref="IBar{T}"/>
	public interface IFoo<T, V> : IBar<T> // <- CS1712, but in the tooltip the type description of T is shown.
		where T : notnull
		where V : notnull
	{
		/// <summary>
		/// Gets the V value.
		/// </summary>
		/// <value>
		/// A value.
		/// </value>
		public V ValueOfV { get; }
	}
```

**Additional information:**

- C# 9.0
- Microsoft Visual Studio Community 2019 Version 16.11.1
- [CS1712](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1712)

**Expected Behavior:**
No diagnostic for CS1712.

**Actual Behavior:**
Get false-positive diagnostic for CS1712.

**Comment by anreton** [comment]
When used like this, there is no warning.

```csharp
/// <summary>
/// Represents a foo abstraction.
/// </summary>
/// <typeparam name="T">
/// <inheritdoc cref="IBar{T}"/>
/// </typeparam>
/// <typeparam name="V">
/// The type of the V.
/// </typeparam>
public interface IFoo<T, V> : IBar<T> // <- CS1712 for T, but in the tooltip the type description of T is shown correctly.
	where T : notnull
	where V : notnull
{
	/// <summary>
	/// Gets the V value.
	/// </summary>
	/// <value>
	/// A value.
	/// </value>
	public V ValueOfV { get; }
}
```

**Comment by CyrusNajmabadi** [comment]
Moving to compiler as this is a compiler diagnostic.
